pkg/store/bucket.go

```diff
@@ -961,6 +981,9 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 		s.metrics.cachedPostingsCompressionTimeSeconds.WithLabelValues("decode").Add(stats.cachedPostingsDecompressionTimeSum.Seconds())
 		s.metrics.cachedPostingsOriginalSizeBytes.Add(float64(stats.cachedPostingsOriginalSizeSum))
 		s.metrics.cachedPostingsCompressedSizeBytes.Add(float64(stats.cachedPostingsCompressedSizeSum))
+		s.metrics.postingsFetchDuration.Observe(stats.postingsFetchDurationSum.Seconds())
+		s.metrics.chunksFetchDuration.Observe(stats.postingsFetchDurationSum.Seconds())
+		s.metrics.seriesFetchDuration.Observe(stats.seriesFetchDurationSum.Seconds())
```

Comment:
I see the value in exposing timings, but I'm a bit dubious about these timings. These timings are tracked and summed up for every single block, but different blocks are queried concurrently. What you get is the sum of timings across all blocks (which again, are queried concurrently) so from an operation perspective I'm not sure how we can get a real value from these timings because they're influenced by how many blocks you query in a given query.

For example, let's say you have to query 2 blocks (A and B). To fetch postings it take 1s for each block, so the sum is 2s, but since they've been queried concurrently the real time it took was 1s.

Reply:
Good point. I'm gonna mark this as work in-progress and try to improve how to track these stats.

Reply:
That is quite important information yes. However, in fact we have some of this information. Maybe there is a way to correlate this latency with number of postings fetched during that time etc? Would that work? :thinking: 


For me the goal is simple: We want to know if Memcached caching is efficient or not. We want to know if we change object storage if has effect in performance or not. Maybe we want to divide this latency by that number of bytes we fetched overall in given time etc (: There must be some good enough thing we can get from it :thinking: 

Any further ideas @brancz @squat ? :thinking:

Reply:
@bwplotka Since the main goal is to have metric to help us to fine-tune caching layer for index lookups. I have changed the implementation and focused on where index cache actually used. Please have a look at it. And let me know if you have any concerns? (on performance etc.)